### PR TITLE
Support higher degree. Up to 8

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: msm/src/column_env.rs [hunk 1]
```diff
@@ -91,12 +92,33 @@ impl<'a, const N: usize, F: FftField> TColumnEnvironment<'a, F> for ColumnEnviro
         }
     }
 
-    fn column_domain(&self, _col: &Self::Column) -> Domain {
+    fn column_domain(&self, col: &Self::Column) -> Domain {
         // TODO FIXME check this is a tricky variable it should match the evalution in column
```
Comment (dannywillems) [inline msm/src/column_env.rs:96]: This comment is removed in https://github.com/o1-labs/proof-systems/pull/2001. Sorry for the second PR. I realized it after creating the other PRs.

File: msm/src/column_env.rs [hunk 2]
```diff
@@ -91,12 +92,33 @@ impl<'a, const N: usize, F: FftField> TColumnEnvironment<'a, F> for ColumnEnviro
...
+                    Domain::D2
+                } else if self.domain.d4.size == domain_size {
+                    Domain::D4
+                } else if self.domain.d8.size == domain_size {
+                    Domain::D8
+                } else {
+                    panic!("Domain not supported. We do support the following multiple of the domain registered in the environment: 1, 2, 4, 8")
+                }
+            }
+            Self::Column::LookupAggregation
```
Comment (dannywillems) [inline msm/src/column_env.rs:115]: This is implemented later, in https://github.com/o1-labs/proof-systems/pull/1999.

File: msm/src/test/mod.rs [hunk 3]
```diff
@@ -226,4 +229,76 @@ mod tests {
         // TODO: Refactorize code in prover to handle a degug or add an adversarial prover.
         // test_soundness_generic(constraints, witness, domain_size, &mut rng);
     }
+
+    #[test]
+    // X_{0} * (X_{1} * X_{2} * X_{3} + 1)
+    fn test_completeness_degree_four() {
```
Comment (volhovm) [inline msm/src/test/mod.rs:235]: Cool!

**Review by volhovm — APPROVED** [review]
Question for myself/asking in person later: Why do MVLookups require degree 8? (I would've guessed something like 2 or 4 given the `phi(x) (alpha + f(x))` constraints...)

**Pull request merged**